save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/WindowOpenness.kt

```diff
@@ -0,0 +1,57 @@
+package com.saveourtool.save.frontend.utils
+
+import react.StateInstance
+import react.useState
+
+/**
+ * Util class which stores state about window openness
+ *
+ * @param isOpenState [StateInstance] with [Boolean] value
+ */
+class WindowOpenness(
+    private val isOpenState: StateInstance<Boolean>
+) {
+    /**
+     * @return current state of window
+     */
+    fun isOpen(): Boolean {
+        val (isOpen, _) = isOpenState
+        return isOpen
+    }
+
+    /**
+     * Open window
+     */
+    fun openWindow() {
+        setIsOpen(true)
+    }
+
+    /**
+     * @return action to open window
+     */
+    fun openWindowAction(): () -> Unit = { openWindow() }
+
+    /**
+     * Close window
+     */
+    fun closeWindow() {
+        setIsOpen(false)
+    }
+
+    /**
+     * @return action to close window
+     */
+    fun closeWindowAction(): () -> Unit = { closeWindow() }
+
+    private fun setIsOpen(value: Boolean) {
+        val (_, setIsOpenState) = isOpenState
+        setIsOpenState(value)
+    }
+
+    companion object {
+        /**
+         * @return [WindowOpenness] with closed state by default
+         */
+        fun create() = WindowOpenness(useState(false))
```

Comment:
I think this method can be called `useWindowOpenness` because it's effectively a hook - a arapper around `useState`

Reply:
agree, will rename and move it to file level to change `WindowOpenness.useWindowOpenness()` to just `useWindowOpenness()`